### PR TITLE
Adding sysctl, returners and removal of six module

### PR DESCRIPTION
**Pull request by daveed12**
- moved salt.modules.linux_sysctl to hubblestack.modules.linux_sysctl
- moved salt.modules.mac_sysctl to hubblestack.modules.mac_sysctl
- moved salt.modules.openbsd_sysctl to hubblestack.modules.openbsd_sysctl
- moved salt.modules.netbsd_sysctl to hubblestack.modules.netbsd_sysctl
- moved salt.modules.freebsd_sysctl to hubblestack.modules.freebsd_sysctl
- Added salt.returners.\_init_ to hubblestack.returners.\_init_
- Removal of six module

### CODE REVIEW

File: hubblestack/extmods/fileserver/azurefs.py [hunk 1]
```diff
@@ -168,7 +165,7 @@ def serve_file(load, fnd):
     with hubblestack.utils.files.fopen(fpath, 'rb') as fp_:
         fp_.seek(load['loc'])
         data = fp_.read(__opts__['file_buffer_size'])
-        if data and six.PY3 and not salt.utils.is_bin_file(fpath):
+        if data and not salt.utils.is_bin_file(fpath):
```
Comment (goravsingal) [inline hubblestack/extmods/fileserver/azurefs.py:168]: In the code we are moving, every salt dependency should be moved. Unless it is a high-level module like loader, for which we already took tasks.

This dependency has also to be moved to our code. "salt.utils.is_bin_file"

File: hubblestack/extmods/fileserver/roots.py [hunk 2]
```diff
@@ -32,7 +32,6 @@
 import hubblestack.utils.platform
 import hubblestack.utils.stringutils
 import salt.utils.versions
```
Comment (goravsingal) [inline hubblestack/extmods/fileserver/roots.py:34]: Same for this, should be under hubblestack.utils.version

File: hubblestack/extmods/fileserver/s3fs.py [hunk 3]
```diff
@@ -248,7 +246,7 @@ def serve_file(load, fnd):
     with hubblestack.utils.files.fopen(cached_file_path, 'rb') as fp_:
         fp_.seek(load['loc'])
         data = fp_.read(__opts__['file_buffer_size'])
-        if data and six.PY3 and not hubblestack.utils.files.is_binary(cached_file_path):
+        if data and not hubblestack.utils.files.is_binary(cached_file_path):
             data = data.decode(__salt_system_encoding__)
         if gzip and data:
             data = salt.utils.gzip_util.compress(data, gzip)
```
Comment (goravsingal) [inline hubblestack/extmods/fileserver/s3fs.py:252]: Need to check for all such occurrence.
Should move under hubblestack.utils.gzip_util

File: hubblestack/extmods/modules/pulsar.py [hunk 4]
```diff
@@ -24,7 +24,6 @@
 from hubblestack.utils.exceptions import CommandExecutionError
 
 # Import salt libs
-import salt.ext.six
 import salt.loader
```
Comment (goravsingal) [inline hubblestack/extmods/modules/pulsar.py:27]: Should use hubblestack.loader

**Review by goravsingal — CHANGES_REQUESTED** [review]

**Comment by goravsingal** [comment]
Overall changes look ok. Thanks for adding unit tests.

We need to cover the referenced code(salt imports) as well. Since we have not taken separate tasks for these. 
Many of those files/functions are already added. You need to see which are the functions which are not there.

File: hubblestack/extmods/fileserver/roots.py (hunk 2 above)
Comment (daveed12) [reply hubblestack/extmods/fileserver/roots.py]: Is it version or versions? I changed it to versions to match the salt module naming schema.

File: hubblestack/modules/linux_sysctl.py [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+    Linux hosts using systemd 207 or later ignore ``/etc/sysctl.conf`` and only
+    load from ``/etc/sysctl.d/*.conf``. This function will do the proper checks
+    and return a default config file which will be valid for the Minion. Hosts
+    running systemd >= 207 will use ``/etc/sysctl.d/99-salt.conf``.
+    CLI Example:
+    .. code-block:: bash
+        salt -G 'kernel:Linux' sysctl.default_config
+    """
+    if (
+        salt.utils.systemd.booted(__context__)
```
Comment (goravsingal) [inline hubblestack/modules/linux_sysctl.py:50]: salt.utils.systemd is already ported to hubblestack.utils.systemd and it has these two methods as well:
booted, version
Please use that.

File: tests/unittests/modules/test_linux_sysctl.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        }
+        mock_asn_cmd = MagicMock(return_value=asn_cmd)
+        cmd = "sysctl -w net.ipv4.ip_forward=1"
+        mock_cmd = MagicMock(return_value=cmd)
+        mock_run = MagicMock(return_value='sysctl')
+        with patch.dict(
+            linux_sysctl.__salt__,
+            {"cmd.run_stdout": mock_cmd, "cmd.run_all": mock_asn_cmd, "cmd.run": mock_run},
+        ):
+            with patch("salt.utils.files.fopen", mock_open()) as m_open:
```
Comment (goravsingal) [inline tests/unittests/modules/test_linux_sysctl.py:107]: it should be hubblestack.utils.files.open

File: tests/unittests/modules/test_linux_sysctl.py [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+                "retcode": 0,
+                "stderr": "",
+                "stdout": "net.ipv4.ip_forward = 1",
+            }
+            mock_asn_cmd = MagicMock(return_value=asn_cmd)
+
+            sys_cmd = "systemd 208\n+PAM +LIBWRAP"
+            mock_sys_cmd = MagicMock(return_value=sys_cmd)
+
+            with patch("salt.utils.files.fopen", mock_open()) as m_open, patch.dict(
```
Comment (goravsingal) [inline tests/unittests/modules/test_linux_sysctl.py:135]: Same for all these occurrences of salt.xyz

File: tests/unittests/modules/test_mac_sysctl.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        mock_cmd = MagicMock(return_value=cmd)
+        with patch.dict(mac_sysctl.__salt__, {"cmd.run_all": mock_cmd}):
+            self.assertEqual(mac_sysctl.assign("net.inet.icmp.icmplim", 50), ret)
+
+    def test_persist_no_conf_failure(self):
+        """
+        Tests adding of config file failure
+        """
+        read_data = IOError(13, "Permission denied", "/file")
+        with patch("salt.utils.files.fopen", mock_open(read_data=read_data)), patch(
```
Comment (goravsingal) [inline tests/unittests/modules/test_mac_sysctl.py:71]: Same here for all occurrences of salt.xyz

File: tests/unittests/modules/test_linux_sysctl.py [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+"""
+    :codeauthor: jmoney <justin@saltstack.com>
+"""
+
+# Import Python libs
+from __future__ import absolute_import, print_function, unicode_literals
+
+# Import Salt Libs
+import hubblestack.modules.linux_sysctl as linux_sysctl
+import salt.modules.systemd_service as systemd
```
Comment (goravsingal) [inline tests/unittests/modules/test_linux_sysctl.py:11]: this will be hubblestack.modules.systemd_service

**Review by goravsingal — CHANGES_REQUESTED** [review]

**Pull request merged**